### DISCUSSION TITLE
Tags Index - Use header from markdown file as URL link text?

### DISCUSSION THREAD

**Discussion by anna-engle** [squidfunk/mkdocs-material#5627] [Q&A] [answered]
Hi, 

In my Tags index, I see that the URL link text comes from the nav. This becomes kind of hairy if there is a one --> many relationship from tag --> nav URL. For instance, consider:

- "Eventing":
     - "Troubleshooting": eventing/debugging.md
- "Terraform":
     - "Troubleshooting": terraform/debugging.md

If both pages are tagged with "Debugging," the nav titles are what will be used ("Troubleshooting"). They will be in the Tags Index for both pages, which can be confusing to the user.

Is there a way to perhaps use the Markdown header as the URL text instead? I really don't want to change these to something like:

- "Eventing":
     - "Eventing Troubleshooting": eventing/debugging.md
- "Terraform":
     - "Terraform Troubleshooting": terraform/debugging.md

as that makes for an ugly nav. :(

**Comment by squidfunk** [comment, marked as answer]
I think there's currently no option to do that, since the tags index is rendered with what MkDocs chose for the `page.title`, but I need to investigate. Could you please create a new change request with some more information, so we can check if other users have the same problems, so that we can gravitate towards a solution? Also, explain how you would like to imagine this working from an authors perspective. The more information the have the better we can solve the problem.

**Reply by anna-engle** [reply]
Thanks, @squidfunk. I've opened a Feature Request here: #5629
